packages/regl-worldview/src/WorldviewContext.js

```diff
@@ -226,12 +281,64 @@ export class WorldviewContext {
     canvasX: number,
     canvasY: number,
     enableStackedObjectEvents: boolean,
-    maxStackedObjectCount: number
+    maxStackedObjectCount: number,
+    collisionMode2D: boolean
   ): Promise<Array<[MouseEventObject, Command<any>]>> {
     if (!this.initializedData) {
       return new Promise((_, reject) => reject(new Error("regl data not initialized yet")));
     }
 
+    if (collisionMode2D) {
+      const mouseEventsWithCommands = [];
+      const [clickX, clickY] = this.canvasToWorldPoint(canvasX, canvasY);
+
+      const drawCalls = Array.from(this._drawCalls.values()).sort((a, b) => (a.layerIndex || 0) - (b.layerIndex || 0));
+      drawCalls.forEach((drawInput: DrawInput) => {
+        const { children, instance, getChildrenForHitmap } = drawInput;
+        if (!children) {
+          return console.debug(`2d hitmap calculation skipped, no draw children`, drawInput);
+        }
+        if (getChildrenForHitmap) {
+          let hitmapChildren = getChildrenForHitmap(children, () => [], []);
+          if (!Array.isArray(hitmapChildren)) {
+            hitmapChildren = [hitmapChildren];
+          }
+
+          hitmapChildren.forEach((item) => {
```

Comment:
Would be nice if you could make the api more generic so it handles more than spheres and polygons. What do you think about having a prop like `testIs2DCollided(item, {clickedX, clickedY, layerIndex})` and let the consumer decide however they'd like to handle the collision? 

We could also avoid adding `zLayer` by using `layerIndex`. This way, the consumer only need to add a single prop to be able to use the 2D collision detection feature.

Reply:
@vidaaudrey Didn't know about the `layerIndex` prop, yeah that works!